roles/cifmw_cephadm/tasks/configure_object.yml

```diff
@@ -38,21 +38,88 @@
+
+    - name: Get uuid for role 'member'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show member -c id -f value"
+      register: role_member_uuid
+
+    - name: Get uuid for role 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show admin -c id -f value"
+      register: role_admin_uuid
+
+    - name: Get uuid for project 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack project show admin -c id -f value"
+      register: project_admin_uuid
+
+    - name: Get uuid for user 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user show admin -c id -f value"
+      register: user_admin_uuid
+
+    - name: Get uuid for role 'swiftoperator'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show swiftoperator -c id -f value"
+      register: role_swiftoperator_uuid
+
+- name: Check if cifmw_cephadm_certificate path exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_cephadm_certificate }}"
+  register: cert_stat_result
+  when: cifmw_cephadm_certificate | length > 0
+
+- name: Check if cifmw_cephadm_key path exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_cephadm_key }}"
+  register: key_stat_result
+  when: cifmw_cephadm_key | length > 0
+
+- name: Update urischeme based on cert/key
+  ansible.builtin.set_fact:
+    cifmw_cephadm_urischeme: "https"
+  when: cert_stat_result.stat.exists | bool and
+        key_stat_result.stat.exists | bool
+
+- name: Configure object store to use rgw
+  cifmw.general.ci_script:
+    extra_args:
+      KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    output_dir: "/home/zuul/ci-framework-data/artifacts"
+    script: |-
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user {{ user_swift_uuid.stdout }} --project {{ project_service_uuid.stdout }} {{ role_member_uuid.stdout }}
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user {{ user_swift_uuid.stdout }} --project {{ project_service_uuid.stdout }} {{ role_admin_uuid.stdout }}
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne {{ service_swift_uuid.stdout }} public {{ cifmw_cephadm_urischeme }}://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne {{ service_swift_uuid.stdout }} internal {{ cifmw_cephadm_urischeme }}://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --project {{ project_admin_uuid.stdout }} --user {{ user_admin_uuid.stdout }} {{ role_swiftoperator_uuid.stdout }}
   delegate_to: localhost
```

Comment:
These changes worked in my environment in that they ran without any error and configured the endpoints with SSL when I rebased on https://github.com/openstack-k8s-operators/ci-framework/pull/1865/files.

```
sh-5.1$ openstack endpoint list | grep swift
| 49825cf208c74cfe8fbed7e647785b61 | regionOne | swift        | object-store | True    | public    | https://172.18.0.2:8080/swift/v1/AUTH_%(tenant_id)s                      |
| 85ae495135754ec7b5e3b9ecc0522d2b | regionOne | swift        | object-store | True    | internal  | https://172.18.0.2:8080/swift/v1/AUTH_%(tenant_id)s                      |
sh-5.1$ 
```

```
TASK [cifmw_cephadm : Configure ceph object store to use external ceph object gateway _raw_params=configure_object.yml] **********************************************************************
Thursday 13 June 2024  21:38:43 -0400 (0:00:00.040)       0:01:01.063 ********* 
included: /home/zuul/src/github.com/openstack-k8s-operators/ci-framework/roles/cifmw_cephadm/tasks/configure_object.yml for compute-0

TASK [cifmw_cephadm : Check if swift is enabled in deployed controlplane _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }}  get $(oc get oscp -n openstack -o name) -o json| jq .spec.swift.enabled] ***
Thursday 13 June 2024  21:38:43 -0400 (0:00:00.048)       0:01:01.112 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Check if swift endpoint is already created _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint list | grep 'swift.*object-store' | wc -l] ***
Thursday 13 June 2024  21:38:43 -0400 (0:00:00.545)       0:01:01.657 ********* 
fatal: [compute-0 -> localhost]: FAILED! => changed=true 
  cmd: set -o pipefail && oc -n openstack rsh openstackclient openstack endpoint list | grep 'swift.*object-store' | wc -l
  delta: '0:00:02.598644'
  end: '2024-06-13 21:38:46.795802'
  msg: non-zero return code
  rc: 1
  start: '2024-06-13 21:38:44.197158'
  stderr: ''
  stderr_lines: <omitted>
  stdout: '0'
  stdout_lines: <omitted>
...ignoring

TASK [cifmw_cephadm : Display a note about swift deployment msg=WARNING: Swift is deployed and the endpoint exists already, ceph RGW cannot be configured as object store service] ***********
Thursday 13 June 2024  21:38:46 -0400 (0:00:02.828)       0:01:04.486 ********* 
skipping: [compute-0]

TASK [cifmw_cephadm : Get uuid for project 'service' _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack project show service -c id -f value] **********
Thursday 13 June 2024  21:38:46 -0400 (0:00:00.043)       0:01:04.529 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Create swift service, user and roles extra_args={'KUBECONFIG': '{{ cifmw_openshift_kubeconfig }}'}, output_dir=/home/zuul/ci-framework-data/artifacts, script=oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack service create --name swift --description 'OpenStack Object Storage' object-store
oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user create --project {{ project_service_uuid.stdout }} --password {{ cifmw_ceph_rgw_keystone_psw }}  swift
oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role create swiftoperator
oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role create ResellerAdmin] ***
Thursday 13 June 2024  21:38:49 -0400 (0:00:02.511)       0:01:07.040 ********* 
Follow script's output here: /home/zuul/ci-framework-data/logs/ci_script_005_create_swift_service_user_and.log
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Get uuid for user 'swift' _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user show swift -c id -f value] ********************
Thursday 13 June 2024  21:38:58 -0400 (0:00:08.778)       0:01:15.819 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Get uuid for service 'swift' _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack service show swift -c id -f value] **************
Thursday 13 June 2024  21:39:00 -0400 (0:00:02.271)       0:01:18.090 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Get uuid for role 'member' _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show member -c id -f value] ******************
Thursday 13 June 2024  21:39:02 -0400 (0:00:02.235)       0:01:20.325 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Get uuid for role 'admin' _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show admin -c id -f value] ********************
Thursday 13 June 2024  21:39:04 -0400 (0:00:02.167)       0:01:22.493 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Get uuid for project 'admin' _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack project show admin -c id -f value] **************
Thursday 13 June 2024  21:39:06 -0400 (0:00:02.138)       0:01:24.631 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Get uuid for user 'admin' _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user show admin -c id -f value] ********************
Thursday 13 June 2024  21:39:09 -0400 (0:00:02.245)       0:01:26.877 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Get uuid for role 'swiftoperator' _raw_params=set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show swiftoperator -c id -f value] ****
Thursday 13 June 2024  21:39:11 -0400 (0:00:02.111)       0:01:28.989 ********* 
changed: [compute-0 -> localhost]

TASK [cifmw_cephadm : Check if cifmw_cephadm_certificate path exists path={{ cifmw_cephadm_certificate }}] ***********************************************************************************
Thursday 13 June 2024  21:39:13 -0400 (0:00:02.031)       0:01:31.020 ********* 
ok: [compute-0]

TASK [cifmw_cephadm : Check if cifmw_cephadm_key path exists path={{ cifmw_cephadm_key }}] ***************************************************************************************************
Thursday 13 June 2024  21:39:13 -0400 (0:00:00.225)       0:01:31.245 ********* 
ok: [compute-0]

TASK [cifmw_cephadm : Update urischeme based on cert/key cifmw_cephadm_urischeme=https] ******************************************************************************************************
Thursday 13 June 2024  21:39:13 -0400 (0:00:00.208)       0:01:31.454 ********* 
ok: [compute-0]

TASK [cifmw_cephadm : Configure object store to use rgw extra_args={'KUBECONFIG': '{{ cifmw_openshift_kubeconfig }}'}, output_dir=/home/zuul/ci-framework-data/artifacts, script=oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user {{ user_swift_uuid.stdout }} --project {{ project_service_uuid.stdout }} {{ role_member_uuid.stdout }}
oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user {{ user_swift_uuid.stdout }} --project {{ project_service_uuid.stdout }} {{ role_admin_uuid.stdout }}
oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne {{ service_swift_uuid.stdout }} public {{ cifmw_cephadm_urischeme }}://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne {{ service_swift_uuid.stdout }} internal {{ cifmw_cephadm_urischeme }}://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --project {{ project_admin_uuid.stdout }} --user {{ user_admin_uuid.stdout }} {{ role_swiftoperator_uuid.stdout }}] ***
Thursday 13 June 2024  21:39:13 -0400 (0:00:00.033)       0:01:31.487 ********* 
Follow script's output here: /home/zuul/ci-framework-data/logs/ci_script_006_configure_object_store_to_use.log
changed: [compute-0 -> localhost]
```

Reply:
Thanks John for testing it, good to know that it worked.